test/src/asyncStressTest.cpp

```diff
@@ -0,0 +1,242 @@
+
+	setupSceneCube(cubePosition);
+
+	// Print random seed for reproducibility
+	fmt::print(stderr, "Graph Stress Test random seed: {}\n", randomSeed);
+
+	// Setup random graphs and helper structures
+	std::vector<RandomGraph> graphs;
+	std::vector<GraphRun> runs;
+	runs.resize(GRAPH_COUNT);
+	for (int g = 0; g < GRAPH_COUNT; ++g) {
+		graphs.emplace_back(GRAPH_SIZE);
+	}
+
+	for (int e = 0; e < GRAPH_EPOCHS; ++e) {
+		// Randomize graph check order
+		std::vector<int> graphCheckOrder = randomizeIndices(GRAPH_COUNT);
+
+		// Prepare graph-run associated - node check order, when apply changes, and expected*
+		for (int g = 0; g < GRAPH_COUNT; ++g) {
+			auto& run = runs.at(g);
+			auto& graph = graphs.at(g);
+			auto checksBeforeChange = std::uniform_int_distribution(0, GRAPH_QUERIES - 1)(randomGenerator);
+			run.nodeCheckOrder = randomizeIndices(GRAPH_SIZE, GRAPH_QUERIES);
+			run.updateRaysAfterCheckingNode = graph.transformNodes.at(run.nodeCheckOrder.at(checksBeforeChange));
+			run.expectedPointCount = graph.rayCount.x() * graph.rayCount.y();
+			run.expectedTransform = graph.getCumulativeTransform();
+			run.expectedPointWorld = Vec3f{graph.raysXY.x(), graph.raysXY.y(), cubePosition.z() - 1};
+			run.nextNodeCheckOrderIndex = 0;
+		}
+
+		// Run all graphs at once
+		for (int g = 0; g < GRAPH_COUNT; ++g) {
+			EXPECT_RGL_SUCCESS(rgl_graph_run(graphs.at(g).useRaysNode));
+		}
+		ASSERT_FALSE(HasFailure());
+
+		// Iterate over graphs and check a single node at a time.
+		// Repeat until all graphs are checked (each node from run.nodeCheckOrder is checked)
+		bool allGraphsChecked = false;
+		while (!allGraphsChecked)
+		{
+			allGraphsChecked = true; // This will be and-ed with is-checked for all graphs
+
+			// Check a single node from each graph
+			for (auto&& g : graphCheckOrder) {
+				// Useful aliases
+				auto& graph = graphs.at(g);
+				auto& run = runs.at(g);
+				auto& checkedNode = graph.transformNodes.at(run.nodeCheckOrder.at(run.nextNodeCheckOrderIndex));
+
+				// Exit if all nodes in this graph were checked
+				bool allNodesChecked = run.nextNodeCheckOrderIndex == run.nodeCheckOrder.size();
+				if (allNodesChecked) {
+					allGraphsChecked = allGraphsChecked && allNodesChecked;
+					continue;
+				}
+
+				// Check if output size is not affected by graph modifications
+				int pointCount = -1, pointSize = -1;
+				EXPECT_RGL_SUCCESS(rgl_graph_get_result_size(checkedNode, RGL_FIELD_XYZ_F32, &pointCount, &pointSize));
+				EXPECT_THAT(pointCount, Eq(run.expectedPointCount));
+
+				// Get results
+				std::vector<Vec3f> results(pointCount);
+				EXPECT_RGL_SUCCESS(rgl_graph_get_result_data(checkedNode, RGL_FIELD_XYZ_F32, results.data()));
+
+				// Verify results are correctly transformed
+				Vec3f expectedPoint = run.expectedTransform.at(checkedNode) * run.expectedPointWorld;
+				EXPECT_THAT(results.at(0).x(), FloatNear(expectedPoint.x(), EPSILON));
```

Comment:
I'm not sure about, however:
We can think about using only `EXPECT_EQ` in cases of equality.
In Your case, everything is alright, however for future uses in tests, we can eliminate stupid mistakes of wrongly used `EXPECT_THAT`.
According to documentation:
`WARNING: Equality matching via EXPECT_THAT(actual_value, expected_value) is supported, however, note that implicit conversions can cause surprising results. For example, EXPECT_THAT(some_bool, "some string") will compile and may pass unintentionally.`

Reply:
Great point. Replaced all `EXPECT_THAT` with `EXPECT_EQ` and `EXPECT_NEAR.`